packages/trader/src/AppV2/Components/TradeParameters/TradeTypeTabs/trade-type-tabs.tsx

```diff
@@ -3,34 +3,44 @@ import { observer } from 'mobx-react';
 import clsx from 'clsx';
 import { SegmentedControlSingleChoice } from '@deriv-com/quill-ui';
 import { useTraderStore } from 'Stores/useTraderStores';
-import { TRADE_TYPES } from '@deriv/shared';
+import { getTradeTypeTabsList } from 'AppV2/Utils/trade-params-utils';
 
 type TTradeTypeTabsProps = {
     is_minimized?: boolean;
 };
 
 const TradeTypeTabs = observer(({ is_minimized }: TTradeTypeTabsProps) => {
-    const { contract_type, is_turbos, is_vanilla, onChange } = useTraderStore();
-    const tab_list = [
-        { label: 'Up', value: TRADE_TYPES.TURBOS.LONG, is_displayed: is_turbos },
-        { label: 'Down', value: TRADE_TYPES.TURBOS.SHORT, is_displayed: is_turbos },
-        { label: 'Call', value: TRADE_TYPES.VANILLA.CALL, is_displayed: is_vanilla },
-        { label: 'Put', value: TRADE_TYPES.VANILLA.PUT, is_displayed: is_vanilla },
-    ];
-    const options = tab_list.filter(({ is_displayed }) => is_displayed);
-    const selected_item_index = options.findIndex(({ value }) => value === contract_type);
+    const { contract_type, onChange, trade_type_tab, setTradeTypeTab } = useTraderStore();
+    const tab_list = getTradeTypeTabsList(contract_type);
+    const initial_index = tab_list.findIndex(tab =>
+        trade_type_tab ? tab.contract_type === trade_type_tab : tab.value === contract_type
+    );
+    const initial_tab_index = initial_index < 0 ? 0 : initial_index;
+    const [tab_index, setTabIndex] = React.useState(initial_tab_index);
+
     const handleTabChange = (selected_item_index: number) => {
-        onChange({ target: { name: 'contract_type', value: options[selected_item_index].value } });
+        const { contract_type: type, value: trade_type } = tab_list[selected_item_index] ?? {};
+        setTabIndex(selected_item_index);
+        setTradeTypeTab(type);
+        if (trade_type !== contract_type) {
+            onChange({ target: { name: 'contract_type', value: trade_type } });
```

Comment:
this would still trigger active_symbols  call as contract_type is changed right? 

Not seeing anywhere we stop active_symbols call from happening on tab switch

Reply:
it's not part of this PR.